tools/slims.py

```diff
@@ -243,3 +214,100 @@ def find_runtag_from_fastqs(fastq_objects) -> str:
         raise Exception('No run tag found in fastq objects')
     else:
         return runtag_newest
+
+
+def find_secanalysis_and_state(bioinformatics_record, sample_name) -> tuple[int, str]:
+    """
+    Takes a slims object with bioinformatics content and returns a tuple containing
+    the secondary analysis ID and secondary analysis state
+
+    :param bioinformatics_records: slims bioinformatics object
+    :param sample_name: Sample name (for logging purposes)
+    :return: Tuple with secondary analysis ID and secondary analysis state
+    """
+    # Note, this is a bit hacky since right now a bioinfo object could have multiple secondary analyses.
+    # This means that cntn_cstm_secondaryAnalysis is a list
+    sec_analysis = bioinformatics_record.cntn_cstm_secondaryAnalysis.value[0]
+    sec_analysis_state = bioinformatics_record.cntn_cstm_SecondaryAnalysisState.value
+    if not sec_analysis:
+        raise Exception(f'A bioinformatics object without secondary analysis info was found: {sample_name}')
+    elif not sec_analysis_state:
+        raise Exception(
+            f'A bioinformatics object without secondary analysis state was found: {sample_name}')
+
+    return sec_analysis, sec_analysis_state
+
+
+def fetch_bioinformatics_record(content_id: str, secondary_analysis: int) -> 'Records':
+    """
+    Fetch a bioinformatics record from slims given a content ID, content type, and secondary analysis ID
+
+    :param content_id: Content ID
+    :param secondary_analysis: Secondary analysis ID
+    :return: SLIMS bioinformatics object
+    """
+    records = slims.fetch("Content", conjunction()
+                          .add(equals("cntn_id", content_id))
+                          .add(equals("cntn_fk_contentType", 23))  # 23 equals bioinformatics
+                          .add(equals("cntn_cstm_secondaryAnalysis", secondary_analysis)))
+
+    if len(records) > 1:
+        raise IOError(f'Multiple bioinformatics objects found for {content_id}')
```

Comment:
wasn't it possible for a sample to have more than one bioinformatics object?

Reply:
There can be multiple bioinformatics objects, but not with the same sec analysis attached. IN this case the functions looks for a specific key of secondary analysis, so it's good to have this check. However, it might be a bit to general name of the function